### PR TITLE
[GR-1281] Gracefully dealing with missing caches

### PR DESCRIPTION
**Pull request by slazicoicr**
image

### CODE REVIEW

File: application/dash_application/pages.py [hunk 1]
```diff
@@ -29,6 +34,17 @@
 
 # Please do not edit this loop
 for name in pagenames:
-    pages.append(importlib.import_module(prefix + name))
+    try:
+        pages.append(importlib.import_module(prefix + name))
+    except (IOError, OSError):
+        print(traceback.format_exc(), file=sys.stderr)
+        ErrorPage = namedtuple('ErrorPage', 'layout title dataversion page_name init_callbacks')
```
Comment (apmasell) [inline application/dash_application/pages.py]: This should probably go somewhere more top-level.

**Review by apmasell — APPROVED** [review]

File: application/dash_application/pages.py [hunk 2]
```diff
@@ -29,6 +34,17 @@
...
+        pages.append(ErrorPage(
+            lambda x: html.H3("Unexpected Error: Failed to load cache"),
```
Comment (callunity) [inline application/dash_application/pages.py]: When this bug appears in the wild, it'll take a developer to fix it. Can we either add Prometheus monitoring that fires an alert for this case, or add a button for the user to submit a ticket describing the issue?

File: application/dash_application/utility/df_manipulation.py [hunk 3]
```diff
@@ -136,32 +136,9 @@ def normalized_merged(df: DataFrame, merged_cols: List[str]):
 
 
 """
-Open a single instance of each cache, and use copies for the reports.
+qc-etl API keeps loaded caches in memory for fast reloading
```
Comment (callunity) [inline application/dash_application/utility/df_manipulation.py:139]: Out of curiosity, how much memory does this take at present?

Comment (slazicoicr) [reply application/dash_application/utility/df_manipulation.py:139]: I am not sure. `flask` takes 1.2GB on my machine.

**Review by callunity — APPROVED** [review]

File: application/dash_application/pages.py (hunk 2 above)
Comment (avarsava) [reply application/dash_application/pages.py]: Echoing this

File: application/dash_application/pages.py (hunk 1 above)
Comment (slazicoicr) [reply application/dash_application/pages.py]: Done

File: application/dash_application/pages.py (hunk 2 above)
Comment (slazicoicr) [reply application/dash_application/pages.py]: Error view contains a link that will generate a JIRA ticket.

**Pull request merged**